clustered/common/src/main/java/org/ehcache/clustered/common/internal/messages/ExceptionCodec.java

```diff
@@ -91,7 +91,7 @@ public static ClusterException decode(StructDecoder<StructDecoder<Void>> decoder
       element.end();
     }
     arrayDecoder.end();
-    Class clazz = null;
+    Class<?> clazz = null;
```

Comment:
This should be `Class<? extends ClusterException>`, right?

Then the whole block would become:

```java
    Class<? extends ClusterException> clazz = null;
    ClusterException exception;
    try {
      clazz = Class.forName(exceptionClassName).asSubclass(ClusterException.class);
    } catch (ClassNotFoundException e) {
      LOGGER.error("Exception type not found", e);
    }
    exception = getClusterException(message, clazz);
    if (exception == null) {
      exception = new UnknownClusterException(message);
    }
    exception.setStackTrace(stackTraceElements);
    return exception;
  }

  @SuppressWarnings("unchecked")
  private static ClusterException getClusterException(String message, Class<? extends ClusterException> clazz) {
    ClusterException exception = null;
    if (clazz != null) {
      try {
        Constructor<? extends ClusterException> declaredConstructor = clazz.getDeclaredConstructor(String.class);
        exception = declaredConstructor.newInstance(message);
      } catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
        LOGGER.error("Failed to instantiate exception object.", e);
      }
    }
    return exception;
  }
```

Reply:
It can. But there's no way to know that `exceptionClassName` is actually of the right type

Reply:
Thats what `asSubclass(ClusterException.class)` does (fourth line).